log4j-api/pom.xml

```diff
@@ -36,6 +36,10 @@
     <bnd-multi-release>true</bnd-multi-release>
     <!-- Differs from OSGi bundle name -->
     <bnd-module-name>org.apache.logging.log4j</bnd-module-name>
+    <bnd-extra-package-options>
+      <!-- Not exported by most OSGi system bundles, hence we use the system classloader to load `sun.reflect.Reflection` -->
+      !sun.reflect
+    </bnd-extra-package-options>
```

Comment:
Has this problem ever been reported? I feel like we are solving a problem of your C64 that you tinker at nights in your attic.

Reply:
A `sun.reflect;resolution:=optional` declaration has been in `log4j-api` since I joined the team.

We must have lost it, when we switched to BND, because it couldn't detect that we are using `sun.reflect.Reflect`. However after I replaced `LoaderUtil.loadClass` with `Class.forName` the entry reappeared and broke our OSGi tests.